File: algorithms/warmup/solve-me-second/solve-me-second.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>

int solve(int a, int b)
{
	return a + b;
}

void load(FILE * in)
{
	int T;
	int a, b;
	fscanf(in, "%d\n", &T);
	for (int i = 0; i < T; i++) {
		fscanf(in, "%d %d\n", &a, &b);
		printf("%d\n", solve(a, b));
	}
}

int main(int argc, char const *argv[])
{
	FILE *in;
	if (argc == 2 && strcmp(argv[1], "COHADAR") == 0) {
		in = fopen("solve-me-second.in", "r");
		assert(in);
	} else {
		in = stdin;
	}
	load(in);
	return 0;
}

File: algorithms/warmup/solve-me-second/SolveMeSecond.java
import java.util.*;
import java.io.*;

public class SolveMeSecond {

	int solve(int a, int b) {
		return a + b;
	}

	void load(Scanner scanner) {
		int T = scanner.nextInt();
		for (int i = 0; i < T; i++) {
			int a = scanner.nextInt();
			int b = scanner.nextInt();
			System.out.println(solve(a, b));
		}
	}

	public static void main(String[] args) throws Exception {
		Scanner scanner;
		if (args.length == 1 && "COHADAR".equals(args[0])) {
			scanner = new Scanner(new File("solve-me-second.in"));
		} else {
			scanner = new Scanner(System.in);
		}
		new SolveMeSecond().load(scanner);
	}
}
